ee/tabby-db/src/job_runs.rs

```diff
@@ -1,23 +1,23 @@
 use anyhow::Result;
-use chrono::{DateTime, Utc};
 use sqlx::{query, FromRow};
+use tabby_db_macros::query_paged_as;
 
 use super::DbConn;
-use crate::make_pagination_query_with_condition;
+use crate::{DateTimeUtc, DbNullable};
 
 #[derive(Default, Clone, FromRow)]
 pub struct JobRunDAO {
-    pub id: i32,
+    pub id: i64,
     #[sqlx(rename = "job")]
     pub name: String,
-    pub exit_code: Option<i32>,
+    pub exit_code: Option<i64>,
     pub stdout: String,
     pub stderr: String,
-    pub created_at: DateTime<Utc>,
-    pub updated_at: DateTime<Utc>,
+    pub created_at: DateTimeUtc,
+    pub updated_at: DateTimeUtc,
 
     #[sqlx(rename = "end_ts")]
-    pub finished_at: Option<DateTime<Utc>>,
+    pub finished_at: DbNullable<DateTimeUtc>,
```

Comment:
We need this DbNullable only because we created DateTimeUtc, correct?

Reply:
Yes, it doesn't automatically know how to convert `Option<NaiveDateTime>` to `Option<DateTimeUTC>`, and I can't implement `From<Option<NaiveDateTime>>` for `Option<DateTimeUtc>` (and can't do it with `Into` either) due to the orphan rule. I made `DbNullable` which should be able to handle optional conversions for any other custom database types we may need.

Reply:
hmm, interesting. I was wondering how sqlx implements similar concept? for example, there has to be very similar implementation for `DateTime<Utc>` and `Option<DateTime<Utc>>`, why does it gain `Option<>` trait automatically?

Reply:
It uses Option for nullable types, and then uses From/Into to convert to the type of the struct field.

Reply:
Worth noting: We still can't use `DateTime<Utc>` as a db type. sqlx knows how to convert its own types (since it just wraps an already-supported type in `Option`), but it only converts using `From`. This means that it sees the column is nullable, wraps it to create `Option<NaiveDateTime>`, and then expects to be able to convert to our target type from that. We can't implement `From<Option<NaiveDateTime>>` for `Option<DateTimeUtc>`, so we have to create a custom type which we *can* implement that for. It's a weird quirk, to be sure.

Reply:
Let's do a discuss offline - if it's not possible, let's stick with `Option<DateTime<Utc>>`

Reply:
We can't use `Option<DateTime<Utc>>` for the same reason we can't use `DateTime<Utc>`